mapstructure_test.go

```diff
@@ -2482,6 +2484,85 @@ func TestDecoder_MatchName(t *testing.T) {
 	}
 }
 
+func TestTypedNilPostHooks(t *testing.T) {
+	type customType1 int
+	type customType2 float64
+	type configType struct {
+		C *customType1 `mapstructure:"c"`
+		A *customType2 `mapstructure:"a"`
+	}
+
+	for i, marshalledConfig := range []map[string]interface{}{
```

Comment:
The first test (which won't use the decoding hooks) works passes just fine without any changes.